File: Easy/firstOccurence.cpp
// Problem 28

class Solution {
public:
  int strStr(string haystack, string needle) {
    if (needle.length() > haystack.length()) {
      return -1;
    }
    else if (needle.length() == haystack.length()) {
      if (needle == haystack) {
        return 0;
      }
      else {
        return -1;
      }
    }

    string check = "";
    int ans = 0;

    for (int i = 0; i < haystack.length(); i++) {
      check.push_back(haystack[i]);
      for (int j = 0; j < needle.length(); j++) {
        if (check.length() == needle.length()) {
          if (check == needle){
            return ans;
          }
          else {
            check.erase(0, 1);
            ans = ans + 1;
          }
        } 
      }
    }
    if (check == needle) {
      return ans;
    }
    else {
      return -1;
    }
  }
};

File: Easy/firstOccurence.rs
// proble 28

impl Solution {
  pub fn str_str(haystack: String, needle: String) -> i32 {
    if needle.is_empty() {
      return 0;
    }

    if needle.len() > haystack.len() {
      return -1;
    }
    
    let haystack_bytes = haystack.as_bytes();
    let needle_bytes = needle.as_bytes();
    
    for i in 0..=haystack.len() - needle.len() {
      if &haystack_bytes[i..i + needle.len()] == needle_bytes {
        return i as i32;
      }
    }
    
    -1      
  }
}